src/main/java/net/snowflake/ingest/SimpleIngestManager.java

```diff
@@ -448,18 +453,24 @@ public IngestResponse ingestFiles(List<StagedFileWrapper> files, UUID requestId)
    *     construction failure
    */
   public IngestResponse ingestFiles(
-      List<StagedFileWrapper> files, UUID requestId, boolean showSkippedFiles)
+      List<StagedFileWrapper> files,
+      UUID requestId,
+      boolean showSkippedFiles,
+      String additionalUserAgentInfo)
```

Comment:
Can we have an overload to the constructor taking the user agent info instead of adding it to all the public apis? Are there any disadvantages that I am missing?

Reply:
Passing it as constructor makes sense as well. (might be a smaller change)
Reason I did go with this approach was because it was being set at request level.

Reply:
But, I kinda agree with having at manager level instead of changing the apis. Let me do that! Thank you!

Reply:
Done